app/src/main/java/com/infomaniak/drive/views/PreviewHeaderView.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Infomaniak kDrive - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.drive.views
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.isGone
+import androidx.transition.AutoTransition
+import androidx.transition.TransitionManager
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.button.MaterialButton
+import com.google.android.material.chip.Chip
+import com.infomaniak.drive.R
+import com.infomaniak.drive.databinding.PreviewHeaderViewBinding
+import com.infomaniak.lib.core.utils.setMargins
+import kotlin.math.max
+
+class PreviewHeaderView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    val pageNumberChip: Chip
+        get() = binding.pageNumberChip
+
+    val openWithButton: MaterialButton
+        get() = binding.openWithButton
+
+    val editButton: MaterialButton
+        get() = binding.editButton
```

Comment:
For openWith and edit button, no problem since I only hide/show them in specific cases. But I need to get the page number chip and since we're using bindings, I don't see how we can get this view without exposing it like this.